pkg/kubectl/cmd/config/set_test.go

```diff
@@ -0,0 +1,106 @@
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package config
+
+import (
+	"bytes"
+	"io/ioutil"
+	"os"
+	"testing"
+
+	"k8s.io/client-go/tools/clientcmd"
+	clientcmdapi "k8s.io/client-go/tools/clientcmd/api"
+)
+
+type setConfigTest struct {
+	description    string
+	config         clientcmdapi.Config
+	args           []string
+	expected       string
+	expectedConfig clientcmdapi.Config
+}
+
+func TestSetConfigContext(t *testing.T) {
+	conf := clientcmdapi.Config{
+		Kind:       "Config",
+		APIVersion: "v1",
+		Clusters: map[string]*clientcmdapi.Cluster{
+			"minikube":   {Server: "https://192.168.99.100:8443"},
+			"my-cluster": {Server: "https://192.168.0.1:3434"},
+		},
+		Contexts: map[string]*clientcmdapi.Context{
+			"minikube":   {AuthInfo: "minikube", Cluster: "minikube"},
+			"my-cluster": {AuthInfo: "mu-cluster", Cluster: "my-cluster"},
+		},
+		CurrentContext: "minikube",
+	}
+	test := setConfigTest{
+		description: "Testing for kubectl config set current-context",
+		config:      conf,
+		args:        []string{"current-context", "my-cluster"},
+		expected:    `Property "current-context" set.` + "\n",
+		expectedConfig: clientcmdapi.Config{
+			Kind:       "Config",
+			APIVersion: "v1",
+			Clusters: map[string]*clientcmdapi.Cluster{
+				"minikube":   {Server: "https://192.168.99.100:8443"},
+				"my-cluster": {Server: "https://192.168.0.1:3434"},
+			},
+			Contexts: map[string]*clientcmdapi.Context{
+				"minikube":   {AuthInfo: "minikube", Cluster: "minikube"},
+				"my-cluster": {AuthInfo: "mu-cluster", Cluster: "my-cluster"},
+			},
+			CurrentContext: "my-cluster",
+		},
+	}
+	test.run(t)
+}
+
+func (test setConfigTest) run(t *testing.T) {
+	fakeKubeFile, err := ioutil.TempFile("", "")
+	if err != nil {
+		t.Fatalf("unexpected error: %v", err)
+	}
+	defer os.Remove(fakeKubeFile.Name())
+	err = clientcmd.WriteToFile(test.config, fakeKubeFile.Name())
+	if err != nil {
+		t.Fatalf("unexpected error: %v", err)
+	}
+	pathOptions := clientcmd.NewDefaultPathOptions()
+	pathOptions.GlobalFile = fakeKubeFile.Name()
+	pathOptions.EnvVar = ""
+	buf := bytes.NewBuffer([]byte{})
+	cmd := NewCmdConfigSet(buf, pathOptions)
+	cmd.SetArgs(test.args)
+	if err := cmd.Execute(); err != nil {
+		t.Fatalf("unexpected error executing command: %v", err)
+	}
+	config, err := clientcmd.LoadFromFile(fakeKubeFile.Name())
+	if err != nil {
+		t.Fatalf("unexpected error loading kubeconfig file: %v", err)
+	}
+	if len(test.expected) != 0 {
+		if buf.String() != test.expected {
+			t.Errorf("Failded in:%q\n expected %v\n but got %v", test.expected, buf.String())
+		}
+	}
+	if len(test.args) == 2 {
+		if test.args[1] != config.CurrentContext {
```

Comment:
This means that the test struct only works for testing set against the current context right?  That's ok, but its more restrictive than the name suggests.  How about comparing against the `expectedConfig` instead?